### DISCUSSION TITLE
Hosting possibilities for gatsby app that does data fetching

### DISCUSSION THREAD

**Discussion by dsfrederic** [gatsbyjs/gatsby#27887] [Help] [answered]
Hi, 

I would like to know which considerations I should make for hosting gatsby and especially for an application that requires data fetching at runtime as done so in this example https://github.com/gatsbyjs/gatsby/tree/master/examples/data-fetching

@LekoArts Can you provide me with some solutions please 

Thanks in advance!

**Comment by herecydev** [comment, marked as answer]
Gatsby isn't doing anything unique for data fetching that other javascript applications are doing. So really the same rules & challenges apply. If you want a unified approach you could use something like [Azure Static Web apps](https://azure.microsoft.com/en-gb/services/app-service/static/) or [Netlify functions](https://www.netlify.com/products/functions/).

If your backend is hosted on a different domain then you may have CORS to setup. But otherwise theirs not really any additional complexity.

Your general considerations will be around security, scalability, consistency, ease of deployment, etc. These are all massive topics though!

**Reply by dsfrederic** [reply]
Okay thank you for you response 👍  I'll try out azure static web apps since my backend is running on Azure as well!

**Comment by wardpeet** [comment]
Like @herecydev mentioned, you can host it everywhere. For dynamic data fetching, you'll need an API server that can give you the data. In the jamstack, mostly serverless API functions are used like azure function, AWS lambda, ...
